src/categorical_algebra/FinSets.jl

```diff
@@ -278,7 +282,10 @@ function Base.show(io::IO, f::FinDomFunctionVector)
   print(io, ")")
 end
 
-force(f::FinDomFunction{Int}) = FinDomFunctionVector(map(f, dom(f)), codom(f))
+function force(f::FinDomFunction{Int})
+  d = map(f, dom(f))
+  FinDomFunctionVector(Vector{eltype(codom(f))}(d), codom(f))
```

Comment:
`FinDomFunctionVector(eltype(codom(f))[ f(x) for x in dom(f) ], codom(f))`.

Actually, instead of doing this in `force`, write `eltype(codom(f))[ f(x) for x in dom(f) ]` when you need the vector to have certain type.